enterprise/kernel/src/main/java/org/neo4j/kernel/impl/enterprise/configuration/EnterpriseEditionSettings.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (c) 2002-2016 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.kernel.impl.enterprise.configuration;
+
+
+import java.util.List;
+
+import org.neo4j.graphdb.config.Setting;
+import org.neo4j.graphdb.factory.Description;
+import org.neo4j.kernel.configuration.Settings;
+
+/**
+ * Enterprise edition specific settings
+ */
+public class EnterpriseEditionSettings
+{
+    @Description( "Specified names of id types (comma separated) that should be reused. " +
+                  "Currently only 'RELATIONSHIP' type is supported. " )
+    public static Setting<List<String>> idTypesToReuse =
+            Settings.setting( "dbms.ids.reuse.types.override", Settings.STRING_LIST, "" );
```

Comment:
I think the name and namespaces of this setting are a little weird. Especially the `.override` part. It could have been just `dbms.id_reuse`